pkg/provisioner/ironic/raid.go

```diff
@@ -12,12 +13,22 @@ import (
 	"github.com/pkg/errors"
 )
 
+const (
+	noRAIDInterface       string = "no-raid"
+	softwareRAIDInterface string = "agent"
+)
+
 // setTargetRAIDCfg set the RAID settings to the ironic Node for RAID configuration steps
-func setTargetRAIDCfg(p *ironicProvisioner, ironicNode *nodes.Node, data provisioner.PrepareData) (err error) {
+func setTargetRAIDCfg(p *ironicProvisioner, raidInterface string, ironicNode *nodes.Node, data provisioner.PrepareData) (err error) {
+	err = checkRAIDConfigure(raidInterface, data.TargetRAIDConfig)
+	if err != nil {
```

Comment:
@Hellcatlk I've just noticed here a latest moment issue that requires to be fixed (hopefully it's something pretty quick).
In case of failure of `checkRAIDConfigure` (or immediately later `BuildTargetRAIDCfg`) a simple error is returned and then managed as a `transientError` in the `startManualCleaning`.
This does not seem the recommended approach in case of misconfiguration: for example, providing a RAID configuration for an host that does not support RAID will result in the BMH stuck in the `Preparing` state without any error set, but just with a stack trace in the log, which is not optimal for the end user.
So I'd suggest to refactor the `setTargetRAIDCfg` method to return directly a `(res provisioner.Result, err error)`, so that any failure for the `checkRAIDConfigure` (or `BuildTargetRAIDCfg`) could be interpreted as `operationFailed`, whereas an eventual issue from `nodes.SetRAIDConfig` will remain as a `transientError`.
So, in case of wrong configuration settings, yhis will give the user directly enough information to fix the problem (without requiring to check the logs)

Reply:
Thanks, done.